buildchain/dodo.py

```diff
@@ -23,29 +25,92 @@
 from buildchain.vagrant import *
 
 
-# mypy doesn't know the type of `doit.reporter.ConsoleReporter`.
-class PrivateReporter(doit.reporter.ConsoleReporter):  # type: ignore
-    """A custom reporter that display public AND private tasks."""
-    desc = 'console, display both public and private tasks'
+# mypy doesn't know the type of `doit.reporter.JsonReporter`.
+class CustomReporter(doit.reporter.JsonReporter):  # type: ignore
+    """A custom reporter that display a JSON object for each task."""
+    desc = 'console, display the execution of each task, generate a build log'
+
+    def __init__(self, outstream: TextIO, options: Dict[str, Any]):
+        super().__init__(outstream, options)
+        self.failures : List[Dict[str, Any]] = []
+        self.runtime_errors : List[str] = []
 
     def execute_task(self, task: doit.task.Task) -> None:
         """Called when a task is executed."""
-        # Ignore tasks that do not define actions.
-        if task.actions:
-            self.write('.  {}\n'.format(task.title()))
+        super().execute_task(task)
```

Comment:
I'm sure I'm missing something here, but why do we call `super().everything` when we previously just printed?

Reply:
The `super` calls the `JsonReporter` methods, which populate an list or dict to keep track of the status of each tasks, the elapsed time, …
This is then used to compute the final JSON at the end (see https://github.com/pydoit/doit/blob/master/doit/reporter.py#L191).

Reply:
Ah right whereas we just hooked into the calls for console since we just needed printing. Thanks